ballerina-tests/np-test-commons/main.bal

```diff
@@ -24,9 +24,17 @@ service /llm on new http:Listener(8080) {
         anydata content = message["content"];
         string contentStr = content.toString();
         test:assertEquals(message.role, "user");
-        test:assertEquals(content, getExpectedPrompt(content.toString()));
+        ChatCompletionTool[]? tools = payload?.tools;
+        if tools is () {
+            test:assertFail("No tools in the payload");
+        }
+
+        FunctionParameters? parameters = tools[0].'function.parameters;
+        if parameters is () {
+            return error("No completion message");
+        }
+        test:assertEquals(parameters, getExpectedParameterSchema(contentStr));
```

Comment:
Why have we removed asserting the prompt?

Reply:
Theres no value of asserting the prompt  guess.
Because we do not change it

Reply:
That is not correct. We construct the prompt string from the natural expression content. This includes inclusions, handling escaped characters, etc. and the tests covered that too.

Even if it was just a string, given that it is not the user directly calling the client method, it's user input that we need to ensure gets used in the expected way.